dfir_lang/src/graph/ops/join.rs

```diff
@@ -134,28 +135,41 @@ pub const JOIN: OperatorConstraints = OperatorConstraints {
             quote_spanned!(op_span=>)
         };
 
-        let mut make_joindata = |persistence, side| {
+        let mut make_joindata = |persistence, in_loop, side| {
             let joindata_ident = wc.make_ident(format!("joindata_{}", side));
             let borrow_ident = wc.make_ident(format!("joindata_{}_borrow", side));
-            let reset = match persistence {
-                Persistence::Tick => quote_spanned! {op_span=>
+            let reset = match (in_loop, persistence) {
+                (false, Persistence::Tick) => quote_spanned! {op_span=>
                     #df_ident.set_state_tick_hook(#joindata_ident, |rcell| #root::util::clear::Clear::clear(rcell.get_mut()));
                 },
-                Persistence::Static => Default::default(),
-                Persistence::Mutable => {
+                (true, Persistence::Tick) => Default::default(),
+                (false, Persistence::Static) => Default::default(),
+                (true, Persistence::Static) => {
                     diagnostics.push(Diagnostic::spanned(
                         op_span,
                         Level::Error,
-                        "An implementation of 'mutable does not exist",
+                        "`'static` is not allowed within loops.",
```

Comment:
I guess eventually we'll have `'loop` which will subsume this, with `'static` becoming the only option at the top-level.